### DISCUSSION TITLE
Can I set low vram usage in controlnet, like in a1111's UI extension?

### DISCUSSION THREAD

**Discussion by okolenmi** [Comfy-Org/ComfyUI#1177] [Q&A] [answered]
I have only 4GB vram. For an average 1200*800 image generation VRAM usage is on 70% level, but with controlnet it's immediately 100%. As soon as the excess memory gets into the shared video memory, a critical error occurs and the generation stops. Is there any solution for my situation, or it's impossible to use it witg 4GB in ComfyUI?

**Comment by lord-lethris** [comment]
you can run ComfyUI with --lowram like this:

`python main.py  --lowvram --preview-method auto --use-split-cross-attention`

Also try increasing your PC's swap file size.
Normally it is common practice with low RAM to have the swap file at 1.5 x Your RAM...  
...but I've learnt from other forums that if your running AI stuff - create a swap file of at least 2 to 3 x your RAM regardless of common rules.  If you have a fast 3.1 USB Stick - add the swap file to that.

**Reply by okolenmi** [reply]
I have 24GB RAM, It's enough as I can see on task manager. Problem in VRAM. I have almost no shared memory usage. When my VRAM memory trying to flow into shared memory I have an error (comfyUI error). I'm already running on low vram mode of comfyui. I'm just not sure if it works for controlnet.

**Reply by lord-lethris** [reply, marked as answer]
ok.  that's odd - you shouldn't be having issues at all.

have you tried re-installing or updating ControlNet?

**Reply by okolenmi** [reply]
Do you mean reinstall of comfyui (It looks like controlnet nodes are default for this UI)? I can try. But not today.

**Reply by lord-lethris** [reply]
You can use the manager to uninstall and re-install ControlNet processors.

**Reply by okolenmi** [reply]
Hmmm. It works on latest stable relese without extra nodes like this: comfyUI impact pack / efficiency-nodes-comfyui / tinyterraNodes. In this case during generation vram memory doesn't flow to shared memory. My limit of resolution with controlnet is about 900*700 images.
![изображение](https://github.com/comfyanonymous/ComfyUI/assets/55319814/744c2f6d-2c85-4d64-8121-aebdb8b49851)

**Reply by Sadpanda07** [reply]
@lord-lethris where to put this command?

**Comment by lord-lethris** [comment]
... you also shouldnt be running 1200*800 images off the cuff with 4gb of RAM.

Ideally you need to Run 768x512, then upscale by 2.  
Use [Ultimate Upscaler](https://github.com/Coyote-A/ultimate-upscale-for-automatic1111) with Upscaler Model [4x-Ultrasharp](https://openmodeldb.info/models/4x-UltraSharp) for better results at low ram.

(_credit to [Kim2091](https://openmodeldb.info/users/kim2091) for [4x-Ultrasharp](https://openmodeldb.info/models/4x-UltraSharp)_)

**Reply by okolenmi** [reply]
I will try it. Thank you. Anyway, controlnet can't run even with 400*400 -_-

**Reply by okolenmi** [reply]
This upscaler blows up my vram memory and then error happens. I will try it later exactly with 4x-Ultrasharp (that was a launch with RealESRGAN_x4).

**Comment by lord-lethris** [comment]
what's your workflow?

**Reply by okolenmi** [reply]
It was simple workflow like this https://github.com/comfyanonymous/ComfyUI_examples/tree/master/controlnet

**Comment by comfyanonymous** [comment]
This shouldn't be a problem anymore on latest ComfyUI. You can run it without any arguments and controlnets should work even on low vram.

**Reply by Sadpanda07** [reply]
It takes too much time to generate image on lowvram.How to enable highvram for faster generation?

**Reply by lord-lethris** [reply]
edit your "start.bat" to contain:

```
@echo off

set PYTHON=
set GIT=

python main.py --gpu-only --highvram --preview-method auto --use-split-cross-attention --force-fp32
```

I found it in another thread :)

**Reply by ROHITchin995** [reply]
@lord-lethris I am not able to locate start.bat
